app/models/manageiq/providers/openstack/network_manager/floating_ip.rb

```diff
@@ -1,2 +1,105 @@
 class ManageIQ::Providers::Openstack::NetworkManager::FloatingIp < ::FloatingIp
+  include ProviderObjectMixin
+  include AsyncDeleteMixin
+  include SupportsFeatureMixin
+
+  supports :create_floating_ip
+  supports :delete_floating_ip
+  supports :update_floating_ip
+
+  def self.raw_create_floating_ip(ext_management_system, options)
+    cloud_tenant = options.delete(:cloud_tenant)
+    floating_ip = nil
+    floating_network_id = CloudNetwork.find(options[:cloud_network_id]).ems_ref
+
+    raw_options = remapping(options)
+
+    ext_management_system.with_provider_connection(connection_options(cloud_tenant)) do |service|
+      floating_ip = service.create_floating_ip(floating_network_id, raw_options)
+    end
+    {:ems_ref => floating_ip['id'], :name => options[:name]}
+  rescue => e
+    _log.error "floating_ip=[#{options[:name]}], error: #{e}"
+    raise MiqException::MiqFloatingIpCreateError, e.to_s, e.backtrace
+  end
+
+  def self.remapping(options)
+    new_options = options.dup
+    new_options[:floating_ip_address] = options[:address] if options[:address]
+    new_options[:tenant_id] = CloudTenant.find(options[:cloud_tenant_id]).ems_ref if options[:cloud_tenant_id]
+    new_options[:port_id] = options[:network_port_ems_ref] if options[:network_port_ems_ref]
+    new_options.delete(:address)
+    new_options.delete(:cloud_network_id)
+    new_options.delete(:network_port_ems_ref)
+    new_options.delete(:cloud_tenant_id)
+    new_options
+  end
+
+  def raw_delete_floating_ip
+    ext_management_system.with_provider_connection(connection_options(cloud_tenant)) do |service|
+      service.delete_floating_ip(ems_ref)
+    end
+  rescue => e
+    _log.error "floating_ip=[#{name}], error: #{e}"
+    raise MiqException::MiqFloatingIpDeleteError, e.to_s, e.backtrace
+  end
+
+  def delete_floating_ip_queue(userid)
+    task_opts = {
+      :action => "deleting Floating IP for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'delete_floating_ip',
```

Comment:
delete_floating_ip is a method from a base class calling the _raw method. Was that already merged?